ngraph/core/reference/include/ngraph/runtime/reference/bucketize.hpp

```diff
@@ -0,0 +1,65 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T, typename P>
+            void bucketize(const T* data,
+                           const T* buckets,
+                           P* out,
+                           const Shape& data_shape,
+                           const Shape& buckets_shape,
+                           bool with_right_bound)
```

Comment:
why is it added into 'runtime/reference' ? The operation is not used in constant folding.

Reply:
There are two different ways to call for a reference implementation:
- evaluate() member function of the op class (constant folding)
- evaluates_map.cpp
Both cases perform call of reference implementation of the op which resides within runtime/reference.
It is correct that this op is not used in constant folding optimization pass, for this reason ref impl is called through evaluates_map, however the reference implementation may not be implemented as part of a test dir.